### ISSUE TITLE
Input checks for pvsamv1 result in inputs required for disabled subarrays

### ISSUE THREAD

**Issue by cpaulgilman** [NREL/ssc#1037]
Simulation error when `subarray2_track_mode` is not assigned even when `subarray2_enable = 0`:

```
Error: check fail: reason unassigned referenced, with 'subarray2_track_mode' for: subarray2_monthly_tilt time -1.000000
EXEC_FAIL: pvsamv1
```
To replicate, run this script in SDKTool with SSC Version 280:

[pv-subarray-input-tests.zip](https://github.com/NREL/ssc/files/12045024/pv-subarray-input-tests.zip)

See https://github.com/NREL/pysam/issues/155

**Comment by caseyzak24** [comment]
@janinefreeman I assume this will be resolved in the next release. Do you have a sense of when that will be? Thanks!

**Comment by janinefreeman** [comment]
Hi @caseyzak24 ! We are hoping for a release in mid-November.

**Comment by cpaulgilman** [comment]
The issue here is that `subarray2_track_mode` is required for some input variable checks even when `subarray2_enable` is set to zero. 

For example, in order to check that `subarray2_tilt` is not set for the 2-axis or seasonal tilt options, the [line in the var table](https://github.com/NREL/ssc/blob/e12327536bed04258632ca66ea3b5aec6118d2ef/ssc/cmod_pvsamv1.cpp#L165C1-L165C498) for `subarray2_track_mode` needs to check `subarray2_track_mode`:

```
{SSC_INPUT, SSC_NUMBER, "subarray2_tilt", "Sub-array 2 Tilt", "degrees", "0=horizontal,90=vertical", "System Design", "subarray2_enable=1&subarray2_track_mode~2&subarray2_track_mode~4", "MIN=0,MAX=90", "" },
```
In other words, for this variable check to pass the following has to be true:

```
subarray2_enable == 1 && subarray2_track_mode != 2 && subarray2_track_mode != 4
```
So, although `subarray2_track_mode` is not required for the simulation, it is required for the variable check logic. Given the way; the variable table is set up, I'm not sure there is a way to avoid this, but I will investigate further.

**Comment by sjanzou** [comment]
Note, that in compute_module::check_required the Boolean expressions are evaluated from left to right. Therefore, if the first constraint fails, the remaining constraints are not checked.  
<img width="1334" alt="image" src="https://github.com/NREL/ssc/assets/6498311/18ba4689-977e-4739-8376-955d74ebfe55">

The example file has a failure for “subarray2_monthly_tilt” because only subarray_track_mode is checked and not subarray2_enable=1 first
<img width="1396" alt="image" src="https://github.com/NREL/ssc/assets/6498311/819c497c-0b3c-42e8-ad16-ea4be608436c">

Pull request #1062 addresses the REQUIRED_IF issue for all of the "subarrayn_monthly_tilt" inputs

**Comment by cpaulgilman** [comment]
Thanks, @sjanzou.
